// validation-br/dist/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.isNUP17 = exports.isTituloEleitor = exports.isRenavam = exports.isPostalCode = exports.isPIS = exports.isJudicialProcess = exports.isCPF = exports.isCNPJ = exports.isCNH = void 0;
const cnh_1 = require("./cnh");
const cnpj_1 = require("./cnpj");
const cpf_1 = require("./cpf");
const judicialProcess_1 = require("./judicialProcess");
const nup17_1 = require("./nup17");
const pisPasep_1 = require("./pisPasep");
const postalCode_1 = require("./postalCode");
const renavam_1 = require("./renavam");
const tituloEleitor_1 = require("./tituloEleitor");
exports.isCNH = (value) => cnh_1.validate(value);
exports.isCNPJ = (value) => cnpj_1.validate(value);
exports.isCPF = (value) => cpf_1.validate(value);
exports.isJudicialProcess = (value) => judicialProcess_1.validate(value);
exports.isPIS = (value) => pisPasep_1.validate(value);
exports.isPostalCode = (value) => postalCode_1.validate(value);
exports.isRenavam = (value) => renavam_1.validate(value);
exports.isTituloEleitor = (value) => tituloEleitor_1.validate(value);
exports.isNUP17 = (value) => nup17_1.validate(value);
exports.default = {
    isCNH: exports.isCNH,
    isCNPJ: exports.isCNPJ,
    isCPF: exports.isCPF,
    isJudicialProcess: exports.isJudicialProcess,
    isPIS: exports.isPIS,
    isPostalCode: exports.isPostalCode,
    isRenavam: exports.isRenavam,
    isTituloEleitor: exports.isTituloEleitor,
    isNUP17: exports.isNUP17,
};
